### DISCUSSION TITLE
MaxListenersExceededWarning and EventTarget

### DISCUSSION THREAD

**Discussion by noseratio** [nodejs/help#4766] [Q&A] [answered]
I think it'd be great to have a programmatic way of disabling the `MaxListenersExceededWarning` warning for the newly introduced `EventTarget` (#35495). I've encountered that while playing with `AbortSignal` in Node v15.3.0, but it can be reproduced with any `EventTarget`:

```JavaScript
const ac = new AbortController();
for (let i = 0; i < 100; i++ ) {
  ac.signal.addEventListener("abort", e => undefined);
}
```
Running this ([runkit](https://runkit.com/noseratio/maxlistenersexceededwarning
)) produces a warning:

`MaxListenersExceededWarning: Possible EventTarget memory leak detected. 11 testEvent listeners added to EventTarget. Use events.setMaxListeners() to increase limit`.

There is no `setMaxListeners` on `EventTarget.prototype`, unless I'm missing something. 

Maybe we should not have this warning on for `EventTarget` by default. Alternatively, it might be nice to have an option for `addEventListener` to control it, something like: 

`signal.addEventListener('abort', handler, { maxListeners: Infinity })`

This way the code could be shared between browsers and Node without mods. 

Thanks again @addaleax and the team for adding `EventTarget` to Node!

**Comment by jasnell** [comment, marked as answer]
See: https://nodejs.org/dist/latest-v15.x/docs/api/events.html#events_eventemitter_setmaxlisteners_n_eventtargets

```js
const {
  setMaxListeners,
  EventEmitter
} = require('events');

const target = new EventTarget();
const emitter = new EventEmitter();

setMaxListeners(5, target, emitter);
```

**Reply by noseratio** [reply]
This is something I've indeed overlooked. Literally it was in the warning 😊
Thanks James!

**Comment by DaAitch** [comment]
For the sake of completeness here is the code for `AbortController` and `setMaxListener`:

```js
const { setMaxListeners } = require('events')

const ac = new AbortController()
const { signal } = ac

// our signal is perhaps going to have 100 listeners (101st listener will again produce a warning)
setMaxListeners(100, signal)

for (let i = 0; i < 100; i++ ) {
  signal.addEventListener('abort', () => {})
}
```

**Reply by fawazahmed0** [reply]
Thanks

**Reply by newTomas** [reply]
I set it to 300 but still get a warning at 101.
